src/grid/src/GridItem.tsx

```diff
@@ -50,36 +60,68 @@ export default defineComponent({
       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
     } = inject(gridInjectionKey)!
     const self = getCurrentInstance()
+
+    const deriveStyle = (): CSSProperties => {
+      // Here is quite a hack, I hope there is a better way to solve it
+      const {
+        privateSpan = defaultSpan,
+        privateShow = true,
+        privateColStart = undefined,
+        privateOffset = 0
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      } = self!.vnode.props as GridItemVNodeProps
+      const { value: xGap } = xGapRef
+      const mergedXGap = pxfy(xGap || 0)
+      return {
+        display: !privateShow ? 'none' : '',
+        gridColumn: `${
+          privateColStart ?? `span ${privateSpan}`
+        } / span ${privateSpan}`,
+        marginLeft: privateOffset
+          ? `calc((100% - (${privateSpan} - 1) * ${mergedXGap}) / ${privateSpan} * ${privateOffset} + ${mergedXGap} * ${privateOffset})`
+          : ''
+      }
+    }
+
+    const createSSRKey = (): string => {
+      return JSON.stringify(deriveStyle())
+    }
+
+    onMounted(() => {
+      const instance = self?.proxy
+      if (!instance) return
+
+      // for SSR, fix bug https://github.com/TuSimple/naive-ui/issues/2462
+      const el: HTMLElement = instance.$el
+      const ssrKey = el.getAttribute(SSR_ATTR_NAME)
+      if (ssrKey) {
+        if (ssrKey !== createSSRKey()) {
+          instance.$forceUpdate()
+        }
+        el.removeAttribute(SSR_ATTR_NAME)
+      }
+    })
+
     return {
+      isSSR: !isBrowser,
       overflow: overflowRef,
       itemStyle: itemStyleRef,
-      deriveStyle: () => {
-        // Here is quite a hack, I hope there is a better way to solve it
-        const {
-          privateSpan = defaultSpan,
-          privateShow = true,
-          privateColStart = undefined,
-          privateOffset = 0
-          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        } = self!.vnode.props as GridItemVNodeProps
-        const { value: xGap } = xGapRef
-        const mergedXGap = pxfy(xGap || 0)
-        return {
-          display: !privateShow ? 'none' : '',
-          gridColumn: `${
-            privateColStart ?? `span ${privateSpan}`
-          } / span ${privateSpan}`,
-          marginLeft: privateOffset
-            ? `calc((100% - (${privateSpan} - 1) * ${mergedXGap}) / ${privateSpan} * ${privateOffset} + ${mergedXGap} * ${privateOffset})`
-            : ''
-        }
-      }
+      deriveStyle,
+      createSSRKey
     }
   },
   render () {
+    const { isSSR, overflow, itemStyle, $slots, deriveStyle, createSSRKey } =
+      this
+
+    const ssrKey = isSSR ? createSSRKey() : undefined
+
     return (
-      <div style={[this.itemStyle as any, this.deriveStyle()]}>
-        {this.$slots.default?.({ overflow: this.overflow })}
+      <div
+        {...{ [SSR_ATTR_NAME]: ssrKey }}
```

Comment:
No need for spread operator.

<img width="1174" alt="image" src="https://user-images.githubusercontent.com/18677354/174467985-93b5ef63-96f0-4b13-b4ac-6d10ae34969a.png">